src/Database/exceptions.php

```diff
@@ -45,4 +45,43 @@ class NotNullConstraintViolationException extends ConstraintViolationException
  */
 class UniqueConstraintViolationException extends ConstraintViolationException
 {
+	/** @var string */
+	private $entry;
+
+	/** @var string */
+	private $key;
+
+
+	/**
+	 * @returns self
+	 */
+	public static function from(\PDOException $src)
+	{
+		$e = parent::from($src);
+
+		preg_match("/Duplicate entry '([^']+)' for key '([^']+)'/", $e->errorInfo[2], $m);
```

Comment:
@Majkl578 Ok, I add result check, but message should be always same, so check shouldn't be needed.

I read your link, but not sure what should I see. Looks like Nette way for handling exception codes is not perfect, but how is this realted to this commit?